modules/vault-cluster/main.tf

```diff
@@ -213,3 +214,27 @@ data "aws_iam_policy_document" "vault_s3" {
     ]
   }
 }
+
+resource "aws_iam_role_policy" "vault_aws_EC2_IAM_Auth" {
+  count  = "${var.enable_EC2_IAM_Auth ? 1 : 0}"
+  name   = "vault_aws_EC2_IAM_Auth"
+  role   = "${aws_iam_role.instance_role.id}"
+  policy = "${element(concat(data.aws_iam_policy_document.vault_aws_EC2_IAM_Auth.*.json, list("")), 0)}"
+}
+
+data "aws_iam_policy_document" "vault_aws_EC2_IAM_Auth" {
+  count = "${var.enable_EC2_IAM_Auth ? 1 : 0}"
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "ec2:DescribeInstances",
+      "iam:GetInstanceProfile",
+      "iam:GetUser",
+      "iam:GetRole",
```

Comment:
Can you add a link to the docs where the requirements for these policies are defined?

Reply:
@brikis98 Done with latest commit  but I leave it to you to 'resolve', if you're satisfied. :) @BMonsalvatge That's exactly where I got them. Thanks for the link. :) Re: cross-account 'sts:AssumeRole': I know the original policies have the sts:AssumeRole stanza, but I'm not sure how to properly enumerate the roles and accounts that's 'generic' for all cases. My present company doesn't really need it - we have sts:AssumeRole added to the IAM Instance Profile at line 175 in this file, and that was enough. However, I added a "TODO", seen in the recent commit. If you have this sussed out, I'd love to see it in the code, too. :)